rockets/clientContext.cpp

```diff
@@ -73,19 +133,33 @@ lws* ClientContext::startHttpRequest(const http::Method method,
     connectInfo.method = http::to_cstring(method);
 #endif
 
+    if (isNoProxyHost(parsedUri.host))
+        disableProxy();
+
     return lws_client_connect_via_info(&connectInfo);
 }
 
 std::unique_ptr<ws::Connection> ClientContext::connect(
     const std::string& uri, const std::string& protocol)
 {
+#if LWS_LIBRARY_VERSION_NUMBER < 2004000
+    const bool recreateVhost = wsProtocolName != protocol;
+#endif
     wsProtocolName = protocol;
     protocols[0].name = wsProtocolName.c_str();
```

Comment:
please also move this hacky reassignement of protocols[0].name into the #if LWS_LIBRARY_VERSION_NUMBER < 2004000, I just re-tested that it is not needed with newer versions which probably correctly use connectInfo.protocol instead

Reply:
Done